src/site/_filters/i18n.js

```diff
@@ -60,6 +60,24 @@ const walk = (dir) => {
 // Scan the _data/i18n directory and create a nested i18n data object
 const data = {i18n: walk(path.join(__dirname, '..', '_data', 'i18n'))};
 
+/**
+ * Flattens the i18n yaml object to contain only translations from given locale.
+ * @param {Object} obj Object with i18n translations (from _data/i18n).
+ * @param {string} locale A locale prefix (example: 'en', 'pl')
+ * @return {Object|string}
+ */
+const getTranslation = function (obj, locale) {
+  const out = {};
+  for (const prop in obj) {
+    if (typeof obj[prop] === 'string') {
```

Comment:
I think this is a bit confusing but I get the intention. It's worth documenting.

- If this object has any keys which point to a string, assume it's a leaf node that has translations.
- Otherwise it's probably an object which nests things, so convert that.

There's a weird corner case where we accidentally set some strings and some objects. I've seen authors do this because they were confused about yaml indentation. But I think this is rare enough not to validate again here.